### PR TITLE
RSDK-7009 - Transform camera schema should not be emptyConfig

### PR DESCRIPTION
**Pull request by zaporter-work**
Ticket: https://viam.atlassian.net/browse/RSDK-7009

The emptyConfig in the transform camera code is causing problems in app. See https://viam.atlassian.net/browse/APP-4204

We need to give each emptyConfig a different name so that the json-schema has different $id values. 

Old generated jsonschema
```json
								{
									"$schema": "https://json-schema.org/draft/2020-12/schema",
									"$id": "https://go.viam.com/rdk/components/camera/transformpipeline/empty-config",
									"$ref": "#/$defs/emptyConfig",
									"$defs": {
										"emptyConfig": {
											"properties": {},
											"additionalProperties": false,
											"type": "object"
										}
									},
									"type": "object",
									"title": "depth_to_pretty",
									"description": "Turns a depth image source into a colorful image, with blue indicating distant points and red indicating nearby points."
								},

``` 							
New generated jsonschema
```json
	{
									"$schema": "https://json-schema.org/draft/2020-12/schema",
									"$id": "https://go.viam.com/rdk/components/camera/transformpipeline/idenity-config",
									"$ref": "#/$defs/idenityConfig",
									"$defs": {
										"idenityConfig": {
											"properties": {},
											"additionalProperties": false,
											"type": "object"
										}
									},
									"type": "object",
									"title": "identity",
									"description": "Does nothing to the image. Can use this to duplicate camera sources, or change the source's stream or parameters."
								},
```

Manually tested with app and it works.

### CODE REVIEW

File: components/camera/transformpipeline/transform.go [hunk 1]
```diff
@@ -32,26 +32,32 @@ const (
...
 	transformTypeIdentity: {
 		string(transformTypeIdentity),
-		&emptyConfig{},
+		&idenityConfig{},
 		"Does nothing to the image. Can use this to duplicate camera sources, or change the source's stream or parameters.",
 	},
 	transformTypeRotate: {
 		string(transformTypeRotate),
-		&emptyConfig{},
+		&rotateConfig{},
```
Comment (zaporter-work) [inline components/camera/transformpipeline/transform.go:60]: is there a reason this had `emptyConfig` before instead of `rotateConfig`? Bug?

Comment (bhaney) [reply components/camera/transformpipeline/transform.go:60]: probably a bug! Thanks for catching it

File: components/camera/transformpipeline/transform.go [hunk 2]
```diff
@@ -32,26 +32,32 @@ const (
...
-type emptyConfig struct{}
-
 // transformRegistration holds pertinent information regarding the available transforms.
 type transformRegistration struct {
 	name        string
 	retType     interface{}
 	description string
 }
 
+// Do not share the share the same config type across multile configs because otherwise
```
Comment (bhaney) [inline components/camera/transformpipeline/transform.go:42]: repeated use of the word `share`

**Review by bhaney — CHANGES_REQUESTED** [review]
nice catch! just a typo and then can merge

File: components/camera/transformpipeline/transform.go (hunk 2 above)
Comment (zaporter-work) [reply components/camera/transformpipeline/transform.go]: Good catch. Fixed in 3fba3a3e8

**Review by bhaney — APPROVED** [review]
LGTM!

**Pull request merged**